src/app/shared/pricing-definitions/pricing-definition/pricing-definition.component.ts

```diff
@@ -0,0 +1,367 @@
+    this.parkingTimeStepUnit = this.parkingTime.controls['stepSizeUnit'];
+    this.initialize();
+  }
+
+  public initialize() {
+    if (this.currentPricingDefinitionID) {
+      // Force refresh the form
+      this.formGroup.updateValueAndValidity();
+      this.formGroup.markAsPristine();
+      this.formGroup.markAllAsTouched();
+      this.loadPricing();
+    }
+  }
+
+  public loadPricing() {
+    if (!this.currentPricingDefinitionID) {
+      return;
+    }
+    this.spinnerService.show();
+    this.centralServerService.getPricingDefinition(this.currentPricingDefinitionID).subscribe((currentPricingDefinition) => {
+      this.spinnerService.hide();
+      this.currentPricingDefinition = currentPricingDefinition;
+      // Init form
+      this.id.setValue(this.currentPricingDefinition.id);
+      this.entityID.setValue(this.currentEntityID);
+      this.entityType.setValue(this.currentEntityType);
+      this.name.setValue(this.currentPricingDefinition.name);
+      this.description.setValue(this.currentPricingDefinition.description);
+      this.validFrom.setValue(this.currentPricingDefinition.staticRestrictions?.validFrom || null);
+      this.validTo.setValue(this.currentPricingDefinition.staticRestrictions?.validTo || null);
+      this.minDate = this.currentPricingDefinition.staticRestrictions?.validFrom || null;
+      this.connectorType.setValue((this.currentPricingDefinition.staticRestrictions?.connectorType) || 'A');
+      this.connectorPowerValue.setValue(this.currentPricingDefinition.staticRestrictions?.connectorPowerkW);
+      this.connectorPowerEnabled.setValue(!!this.connectorPowerValue.value);
+      this.dimensionsMap = currentPricingDefinition.dimensions;
+      this.dimensionsKeys = Object.keys(this.dimensionsMap);
+      this.initializeDimensions();
+    }, (error) => {
+      this.spinnerService.hide();
+      switch (error.status) {
+        case HTTPError.OBJECT_DOES_NOT_EXIST_ERROR:
+          this.messageService.showErrorMessage('settings.pricing.pricing_definition_not_found');
+          break;
+        default:
+          Utils.handleHttpError(error, this.router, this.messageService,
+            this.centralServerService, 'settings.pricing.pricing_definition_error');
+      }
+    });
+  }
+
+  public closeDialog(saved: boolean = false) {
+    if (this.inDialog) {
+      this.dialogRef.close(saved);
+    }
+  }
+
+  public close() {
+    Utils.checkAndSaveAndCloseDialog(this.formGroup, this.dialogService,
+      this.translateService, this.save.bind(this), this.closeDialog.bind(this));
+  }
+
+  public save(pricingDefinition: PricingDefinition) {
+    this.consistencyCheck(pricingDefinition);
+    if (this.currentPricingDefinitionID) {
+      // update existing Pricing Definition
+      this.updatePricingDefinition(pricingDefinition);
+    } else {
+      // create new Pricing Definition
+      this.createPricingDefinition(pricingDefinition);
+    }
+  }
+
+  public toggle(event){
```

Comment:
Toggle what?
Optimized but not readable, not typed and one generic method to handle 8 checkboxes/sliders in the HTTP template is not really nice!
Try to type at least the event and as long as this method is small, I'm okay to keep it as it is ;-).